File: src/org/epstudios/epmobile/CycleLength.java
package org.epstudios.epmobile;

import android.app.Activity;
import android.graphics.Color;
import android.os.Bundle;
import android.view.View;
import android.view.View.OnClickListener;
import android.widget.RadioGroup;
import android.widget.RadioButton;
import android.widget.TextView;
import android.widget.EditText;

public class CycleLength extends Activity implements OnClickListener {
	@Override
	protected void onCreate(Bundle savedInstanceState)  {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.cyclelength);
		
		View calculateQtcButton = findViewById(R.id.calculate_result_button);
        calculateQtcButton.setOnClickListener(this);
        View clearButton = findViewById(R.id.clear_button);
        clearButton.setOnClickListener(this);
        intervalRateRadioGroup = (RadioGroup) findViewById(R.id.intervalRateRadioGroup);
        resultTextView = (TextView) findViewById(R.id.calculated_result);
        inputEditText = (EditText) findViewById(R.id.inputEditText);
        clRadioButton = (RadioButton) findViewById(R.id.cl_button);
        hrRadioButton = (RadioButton) findViewById(R.id.hr_button);
        clRadioButton.setOnClickListener(this);
        hrRadioButton.setOnClickListener(this);
        
        
        setInputHint();
	}
	
	private TextView resultTextView;
	private EditText inputEditText;
	private RadioGroup intervalRateRadioGroup;
	private RadioButton clRadioButton;
	private RadioButton hrRadioButton;

	
	public void onClick(View v) {
		switch (v.getId()) {
		case R.id.calculate_result_button:
			calculateResult();
			break;
		case R.id.clear_button:
			clearEntries();
			break;
		case R.id.cl_button:
		case R.id.hr_button:
			setInputHint();
			break;
		}
	}
	
	private void setInputHint() {
		if (intervalRateRadioGroup.getCheckedRadioButtonId() == R.id.cl_button)
			inputEditText.setHint(R.string.cl_hint);
		else
			inputEditText.setHint(R.string.hr_hint);
			
	}
	
	private void calculateResult() {
		CharSequence resultText = inputEditText.getText();
		resultTextView.setTextColor(Color.GREEN);
		try {
			int result = Integer.parseInt(resultText.toString());
			if (result == 0)
				throw new NumberFormatException();
			result = 60000 / result;
			if (intervalRateRadioGroup.getCheckedRadioButtonId() == R.id.cl_button)
				resultTextView.setText("HR = " + String.valueOf(result) + " bpm");
			else
				resultTextView.setText("CL = " + String.valueOf(result) + " msec");
		}
		catch (NumberFormatException e) {	
			resultTextView.setText("Invalid!");
			resultTextView.setTextColor(Color.RED);
		}		
	}		
	
	
	private void clearEntries() {
		inputEditText.setText(null);
		resultTextView.setText(getString(R.string.calculated_result_label));
		inputEditText.requestFocus();
		setInputHint();
	}
}

File: app/src/main/java/org/epstudios/epmobile/features/calculators/ui/CycleLength.kt
package org.epstudios.epmobile.features.calculators.ui

import android.content.Intent
import android.os.Bundle
import android.view.MenuItem
import org.epstudios.epmobile.R
import org.epstudios.epmobile.core.ui.base.EpActivity
import org.epstudios.epmobile.databinding.CyclelengthBinding
import kotlin.math.roundToInt

class CycleLength : EpActivity() {

    private lateinit var binding: CyclelengthBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = CyclelengthBinding.inflate(layoutInflater)
        setContentView(binding.root)
        setupInsets(binding.cycleLengthRootView)
        initToolbar()

        binding.calculateButtonsLayout.calculateButton.setOnClickListener { calculateResult() }
        binding.calculateButtonsLayout.clearButton.setOnClickListener { clearEntries() }

        binding.intervalRateChipGroup.setOnCheckedChangeListener { _, _ ->
            updateInputHint()
            binding.calculatedResult.setText(R.string.calculated_result_label)
        }
        // Set initial hint
        updateInputHint()
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return when (item.itemId) {
            android.R.id.home -> {
                val parentActivityIntent = Intent(this, CalculatorList::class.java)
                parentActivityIntent.addFlags(
                    Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_ACTIVITY_NEW_TASK
                )
                startActivity(parentActivityIntent)
                finish()
                true
            }
            else -> super.onOptionsItemSelected(item)
        }
    }

    private fun updateInputHint() {
        val hintResId = if (binding.intervalRateChipGroup.checkedChipId == R.id.cl_chip) {
            R.string.cl_hint
        } else {
            R.string.hr_hint
        }
        binding.inputLayout.hint = getString(hintResId)
    }

    private fun calculateResult() {
        val inputText = binding.inputEditText.text.toString()
        binding.calculatedResult.setTextAppearance(R.style.TextAppearance_Calculator_Result)
        try {
            val input = inputText.toInt()
            if (input == 0) throw NumberFormatException()

            val result = calculate(input)
            val resultStringId = if (binding.intervalRateChipGroup.checkedChipId == R.id.cl_chip) {
                R.string.cl_result_as_rate
            } else {
                R.string.cl_result_as_interval
            }
            binding.calculatedResult.text = getString(resultStringId, result.toString())
        } catch (e: NumberFormatException) {
            binding.calculatedResult.text = getString(R.string.invalid_warning)
            binding.calculatedResult.setTextAppearance(R.style.TextAppearance_Calculator_Error)
        }
    }

    private fun clearEntries() {
        binding.inputEditText.text = null
        binding.calculatedResult.setText(R.string.calculated_result_label)
        binding.inputEditText.requestFocus()
    }

    companion object {
        fun calculate(value: Int): Int {
            require(value != 0)
            return (60000.0 / value).roundToInt()
        }
    }
}
